nexus/arch.cc

```diff
@@ -171,6 +171,19 @@ Arch::Arch(ArchArgs args) : args(args)
     }
     if (!speed_grade)
         log_error("Unknown speed grade '%s'.\n", speed.c_str());
+
+    std::unordered_set<IdString> bel_types;
+    for(BelId bel : getBels()) {
+        bel_types.insert(getBelType(bel));
+    }
+
+    for(IdString bel_type : bel_types) {
+        cell_types.push_back(bel_type);
+
+        PartitionId partition;
+        partition.name = bel_type;
+        partitions.push_back(partition);
```

Comment:
as Nexus uses a split-slice arrangement, to match the way that HeAP cellgroups (which are essentially the precursor to partitions) are being set up `OXIDE_FF` and `OXIDE_COMB` should really be combined into one `SLICE` partition

Reply:
I believe I did that here: https://github.com/YosysHQ/nextpnr/pull/557/files#diff-f8b12dcf636347f6473ca026e04cce243869c14dd298d2be18d1801a48f3635dR651-R652

Based on my understanding of placer_heap, the AP should run as:
 - All partitions
 - OXIDE_FF / OXIDE_COMB partitions
 - Each other BEL type / partition by themselves

Did I missing something?

Reply:
> OXIDE_FF and OXIDE_COMB should really be combined into one SLICE partition

I'd argue that because of how occupancy / bel counts are tracked (both before and after this PR), these need to remain as separate partitions because occupancy counts are only really relevant as FF vs COMB, not together.  For example putting all the FF's in one cut and all the COMB's in the other would result in a very poor solution.